html/mycakeapp/src/Controller/MembersController.php

```diff
@@ -15,8 +19,12 @@ class MembersController extends AppController
 {
     public function initialize()
     {
-        $member = null;
-        $this->set('member', $member);
+        parent::initialize();
+        $member = $this->Auth->user();
+        if (!empty($member)) {
+            // コードレビュー時はコメント化してください
+            return $this->redirect(['controller' => 'error']);
+        }
```

Comment:
①ログイン後にログアウトボタンを押すとエラーコントローラーに遷移してしまいます。
```suggestion
        parent::initialize();
        $member = $this->Auth->user();
        if (!empty($member) && $this->request->action !== 'logout') {
            // コードレビュー時はコメント化してください
            return $this->redirect(['controller' => 'error']);
        }
```
②他のコントローラー(main, mypage, reserves)でも同様な形でログインの有無による画面遷移の対応をお願いします
③エラー画面の作成もお願いいたします。(簡単なビュー作成だけなので時間はかからないかと！)

Reply:
確かにログアウトいくとエラー画面に遷移しますね。
修正します！！
同時に他のコントローラーもユーザーの状態によって遷移するよう変更するのと、エラー画面の作成も行います